### PR TITLE
[cli] Prevent regression in loading passphrase secrets provider from state

### PR DESCRIPTION
**Pull request by stack72**
Fixes: #6974

Passphrase Environment variables were set before loading the
secrets provider from state

Unfortunately, it seems that some users are using empty passphrases
and thus this newly introduced logic has broken their usecases

We now check that the environment variables are set - it doesn't
matter if they are set as empty, but the existance of an empty
environment variabe still suggests that it is an intentional
empty passphrase

### CODE REVIEW

**Comment by mikhailshilkov** [comment]
Do you have a link to the PR which introduced the regression?

**Comment by stack72** [comment]
> Do you have a link to the PR which introduced the regression?

https://github.com/pulumi/pulumi/pull/6893

File: pkg/secrets/passphrase/manager.go
Comment (mikhailshilkov) [inline pkg/secrets/passphrase/manager.go:189]: Would `!isFound` be enough?

Comment (stack72) [reply pkg/secrets/passphrase/manager.go:189]: It *could* be but I am super keen to unblock these users - I can test that out and make it simpler once this P1 is fixed up

Comment (stack72) [reply pkg/secrets/passphrase/manager.go:189]: Ok, tested and you are correct that on;y the check for !isFound is necessary here - we don't care about the "" part

Comment (mikhailshilkov) [reply pkg/secrets/passphrase/manager.go:189]: Looking at the code, you never return `false` with a non-empty phrase, it seems. So, it looks like no risk to me. I'll leave it to your judgment.

**Review by mikhailshilkov — APPROVED** [review]

**Comment by thiagosalvatore** [comment]
When is it going to be released? It is impacting our production pipeline

**Comment by joeduffy** [comment]
Should we have a test to ensure empty passphrases continue working in the future?

**Comment by stack72** [comment]
> Should we have a test to ensure empty passphrases continue working in the future?

I can follow up with that - our entire test framework sets one by default so I'll need to pull at that thread to make sure we can do this

I'll open an issue to make sure this is inplace

**Comment by hakoerber** [comment]
Hey @stack72, thank you very much for the super fast fix!

Just pulled `3.2.1`, works like a charm.

**Comment by stack72** [comment]
great news @hakoerber - thanks so much for verifying this for me!

**Pull request merged**
